pkg/build/controller/build/build_controller.go

```diff
@@ -396,34 +417,65 @@ func (bc *BuildController) resolveOutputDockerImageReference(build *buildapi.Bui
 			var ok bool
 			streamName, tag, ok = imageapi.SplitImageStreamTag(streamName)
 			if !ok {
-				return "", fmt.Errorf("the referenced image stream tag is invalid: %s", outputTo.Name)
+				return "", build.Spec.Output.PushSecret, fmt.Errorf("the referenced image stream tag is invalid: %s", outputTo.Name)
```

Comment:
could you return nil instead of build.Spec.Output.PushSecret for at least all cases where you're returning a non-nil error?